### ISSUE TITLE
Build fails due to missing library

### ISSUE THREAD

**Issue by fnky** [nodegui/nodegui-plugin-title-bar#44]
While trying to install the node package, the build fails with a fatal error due to a missing library.

### Actual

Build fails with the following error:

```
Building CXX object CMakeFiles/nodegui_plugin_title_bar.dir/src/main.mm.o
/Users/fnky/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/src/main.mm:2:10: fatal error:
      'QWindow' file not found
#include <QWindow>
         ^~~~~~~~~
1 error generated.
```

CMake's throws a warning about not being able to find the correct Qt version:

```
CMake Warning (dev) in CMakeLists.txt:
  AUTOGEN: No valid Qt version found for target nodegui_plugin_title_bar.
  AUTOMOC, AUTOUIC and AUTORCC disabled.  Consider adding:

    find_package(Qt<QTVERSION> COMPONENTS Widgets)

  to your CMakeLists.txt file.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

This is odd since nodegui does try to find the package in `nodegui:config/qt.cmake`:

https://github.com/nodegui/nodegui/blob/dda2debf36ddcdfeb29772da8f140511db2267f9/config/qt.cmake#L23

Which is is included by `nodegui:config/plugin.cmake`:

https://github.com/nodegui/nodegui/blob/dda2debf36ddcdfeb29772da8f140511db2267f9/config/plugin.cmake#L11

That is referenced from `nodegui:plugin/index.js`:

https://github.com/nodegui/nodegui/blob/dda2debf36ddcdfeb29772da8f140511db2267f9/plugin/index.js#L3-L5

and resolved in `nodegui-plugin-title-bar`s CMakeLists:

https://github.com/nodegui/nodegui-plugin-title-bar/blob/9d0dd273ac86a76327cc1b4d8933871bc1c1c284/CMakeLists.txt#L3-L9

The full output is:

```
> @nodegui/plugin-title-bar@0.0.1 postinstall /Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar
> npm run build:addon


> @nodegui/plugin-title-bar@0.0.1 build:addon /Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar
> cross-env CMAKE_BUILD_PARALLEL_LEVEL=8 cmake-js compile

[
  '/Users/iChrille/.fnm/node-versions/v12.14.1/installation/bin/node',
  '/Users/iChrille/Projects/nodegui-test/node_modules/.bin/cmake-js',
  'compile'
]
info TOOL Using Unix Makefiles generator.
info CMD CONFIGURE
info RUN cmake "/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar" --no-warn-unused-cli -G"Unix Makefiles" -DCMAKE_JS_VERSION="6.1.0" -DCMAKE_BUILD_TYPE="Release" -DCMAKE_LIBRARY_OUTPUT_DIRECTORY="/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/build/Release" -DCMAKE_JS_INC="/Users/iChrille/.cmake-js/node-x64/v12.14.1/include/node;/Users/iChrille/Projects/nodegui-test/node_modules/nan" -DCMAKE_JS_SRC="" -DNODE_RUNTIME="node" -DNODE_RUNTIMEVERSION="12.14.1" -DNODE_ARCH="x64" -DCMAKE_CXX_FLAGS="-D_DARWIN_USE_64_BIT_INODE=1 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DBUILDING_NODE_EXTENSION" -DCMAKE_SHARED_LINKER_FLAGS="-undefined dynamic_lookup"
Not searching for unused variables given on the command line.
-- The C compiler identification is AppleClang 10.0.1.10010046
-- The CXX compiler identification is AppleClang 10.0.1.10010046
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using QT installation for nodegui_plugin_title_bar QT_CMAKE_HOME_DIR:/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/nodegui/miniqt/5.14.1/clang_64/lib/cmake/Qt5
-- Configuring done
CMake Warning (dev) in CMakeLists.txt:
  AUTOGEN: No valid Qt version found for target nodegui_plugin_title_bar.
  AUTOMOC, AUTOUIC and AUTORCC disabled.  Consider adding:

    find_package(Qt<QTVERSION> COMPONENTS Widgets)

  to your CMakeLists.txt file.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/build
info CMD BUILD
info RUN cmake --build "/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/build" --config Release
Scanning dependencies of target nodegui_plugin_title_bar
[ 50%] Building CXX object CMakeFiles/nodegui_plugin_title_bar.dir/src/main.mm.o
/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/src/main.mm:2:10: fatal error: 'QWindow' file not found
#include <QWindow>
         ^~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/nodegui_plugin_title_bar.dir/src/main.mm.o] Error 1
make[1]: *** [CMakeFiles/nodegui_plugin_title_bar.dir/all] Error 2
make: *** [all] Error 2
info REP Build has been failed, trying to do a full rebuild.
info CMD CLEAN
info RUN cmake -E remove_directory "/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/build"
info CMD CONFIGURE
info RUN cmake "/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar" --no-warn-unused-cli -G"Unix Makefiles" -DCMAKE_JS_VERSION="6.1.0" -DCMAKE_BUILD_TYPE="Release" -DCMAKE_LIBRARY_OUTPUT_DIRECTORY="/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/build/Release" -DCMAKE_JS_INC="/Users/iChrille/.cmake-js/node-x64/v12.14.1/include/node;/Users/iChrille/Projects/nodegui-test/node_modules/nan" -DCMAKE_JS_SRC="" -DNODE_RUNTIME="node" -DNODE_RUNTIMEVERSION="12.14.1" -DNODE_ARCH="x64" -DCMAKE_CXX_FLAGS="-D_DARWIN_USE_64_BIT_INODE=1 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DBUILDING_NODE_EXTENSION" -DCMAKE_SHARED_LINKER_FLAGS="-undefined dynamic_lookup"
Not searching for unused variables given on the command line.
-- The C compiler identification is AppleClang 10.0.1.10010046
-- The CXX compiler identification is AppleClang 10.0.1.10010046
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using QT installation for nodegui_plugin_title_bar QT_CMAKE_HOME_DIR:/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/nodegui/miniqt/5.14.1/clang_64/lib/cmake/Qt5
-- Configuring done
CMake Warning (dev) in CMakeLists.txt:
  AUTOGEN: No valid Qt version found for target nodegui_plugin_title_bar.
  AUTOMOC, AUTOUIC and AUTORCC disabled.  Consider adding:

    find_package(Qt<QTVERSION> COMPONENTS Widgets)

  to your CMakeLists.txt file.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/build
info CMD BUILD
info RUN cmake --build "/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/build" --config Release
Scanning dependencies of target nodegui_plugin_title_bar
[ 50%] Building CXX object CMakeFiles/nodegui_plugin_title_bar.dir/src/main.mm.o
/Users/iChrille/Projects/nodegui-test/node_modules/@nodegui/plugin-title-bar/src/main.mm:2:10: fatal error: 'QWindow' file not found
#include <QWindow>
         ^~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/nodegui_plugin_title_bar.dir/src/main.mm.o] Error 1
make[1]: *** [CMakeFiles/nodegui_plugin_title_bar.dir/all] Error 2
make: *** [all] Error 2
ERR! OMG Process terminated: 2
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @nodegui/plugin-title-bar@0.0.1 build:addon: `cross-env CMAKE_BUILD_PARALLEL_LEVEL=8 cmake-js compile`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @nodegui/plugin-title-bar@0.0.1 build:addon script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/iChrille/.npm/_logs/2020-03-15T17_43_22_235Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @nodegui/plugin-title-bar@0.0.1 postinstall: `npm run build:addon`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @nodegui/plugin-title-bar@0.0.1 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/iChrille/.npm/_logs/2020-03-15T17_43_22_689Z-debug.log
```

### Expected behavior

CMake should find the correct package (Qt5) with the correct components and include `QtWindow` correctly from nodegui's packaged Qt library (`@nodegui/nodegui/miniqt/5.14.1`).

### Additional notes

This is possibly a compatibility issue between nodegui and nodegui-plugin-title-bar since this library depends on `@nodegui/nodegui@0.11.0` in development, but accepts any version for peer dependency. I use a newer minor version (`0.15.2`).

### System

- macOS Mojave 10.14.5
- CMake 3.16.4
- Apple LLVM version 10.0.1 (clang-1001.0.46.4) [x86_64-apple-darwin18.6.0]
- `@nodegui/nodegui@0.15.2`

**Comment by jardenliu** [comment]
@master-atul Is this no longer maintained?

**Comment by a7ul** [comment]
@jardenliu  its now available in latest v0.0.2

**Comment by jardenliu** [comment]
thanks
